client.go

```diff
@@ -75,6 +80,15 @@ type Client struct {
 	node       *Node
 	sharedKeys map[string]*[sharedKeySize]byte
 	wallet     *Wallet
+
+	// client auth
+	chChallengeSignature chan stSaltAndSignature
```

Comment:
In general I would suggest we use pointer for better performance (this particular case doesn't have performance issue though)

Reply:
Yes, modifed to pointer.